app/auth/security.py

```diff
@@ -0,0 +1,140 @@
+from typing import Annotated
+from datetime import timezone, timedelta, datetime
+
+from passlib.hash import argon2
+import jwt
+from jwt.exceptions import InvalidTokenError
+from fastapi.security import OAuth2PasswordBearer
+from fastapi import HTTPException, Depends, status
+from app.models.users import Users, TokenData, Token
+from app.config import Config
+from app.db import get_session
+from sqlmodel import Session
+
+import logging
+
+ALGORITHM = "HS256"
+ACCESS_TOKEN_EXPIRE_MINUTES = 30
+SECRET_KEY = Config.SECRET_KEY
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")
+
+
+def verify_password(plain_password, hashed_password):
+    return argon2.verify(plain_password, hashed_password)
+
+
+def get_password_hash(password):
+    """
+    This function runs argon2 salting and hashing via passlib.
+
+    Args:
+        password: Password data as a string.
+
+    Returns:
+        password: Returns a hashed and salted password using
+        passlib argon2.
+    """
+    return argon2.hash(password)
+
+
+def get_user(session, username: str):
+    user = session.get(Users, username)
+
+    return user
```

Comment:
What happens if a I request a username that doesn't exist? Will it error? What object type does it return?

Please could you add a comment explaining this, or remove the function and replace all uses with `session.get(Users, username)`

Many thanks 😄